### PR TITLE
Extra throws in pTest removed

### LINKED ISSUE #842: pTest problem with __del__ and exceptions
I've currently got some failures (on Windows, Python 3.8), which gives me the following confusing output
```
....
.EException ignored in: <function pTest.__del__ at 0x0000015A35299B80>
Traceback (most recent call last):
  File "C:\Users\krisf\Documents\devel\buildVC\SIRF-SuperBuild-python\INSTALL\python\sirf\Utilities.py", line 267, in __del__
    raise ValueError(msg)
ValueError: 1 failures
...
======================================================================
ERROR: tests_four.test_main
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\krisf\Documents\devel\buildVC\SIRF-SuperBuild-python\INSTALL\python\sirf\Utilities.py", line 287, in check
    check_tolerance(expected, value, abs_tol, rel_tol)
  File "C:\Users\krisf\Documents\devel\buildVC\SIRF-SuperBuild-python\INSTALL\python\sirf\Utilities.py", line 240, in check_tolerance
    raise ValueError("|%.4g - %.4g| > %.3g" %
ValueError: |2.13e-08 - -0.001098| > 4.26e-11

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\krisf\miniconda3\envs\sirf\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "C:\Users\krisf\Documents\devel\buildVC\SIRF-SuperBuild-python\sources\SIRF\src\xSTIR\pSTIR\tests\tests_four.py", line 42, in test_main
    test.check(1 - math.sqrt(acq_data.dot(acq_data)) / acq_data.norm())
  File "C:\Users\krisf\Documents\devel\buildVC\SIRF-SuperBuild-python\INSTALL\python\sirf\Utilities.py", line 292, in check
    raise ValueError(msg)
ValueError: +++ test 7 failed:|2.13e-08 - -0.001098| > 4.26e-11
```
I think the underlying reason (certainly for the first warning, possibly not for the double exception) is that `pTest.__del__` should not raise an exception, but exceptions in `__del__` are printed but ignored. In fact, `pTest.__del__` probably shouldn't exist at all as it isn't guaranteed to be called apparently.  See for instance https://stackoverflow.com/questions/1481488/what-is-the-del-method-how-to-call-it

In fact, should `runner` not test for exceptions?
https://github.com/SyneRBI/SIRF/blob/4bce83f0c2049011bc506584894c4be94061b190/src/common/Utilities.py#L340

### PR DESCRIPTION
**Pull request by evgueni-ovtchinnikov**
aims to fix #842

### CODE REVIEW

File: src/common/Utilities.py [hunk 1]
```diff
@@ -231,14 +231,16 @@ def show_3D_array\
     return 0
 
 
-def check_tolerance(expected, actual, abstol=0, reltol=1e-4):
+def check_tolerance(expected, actual, abstol=0, reltol=2e-3):
```
Comment (KrisThielemans) [inline src/common/Utilities.py:234]: intentional to decrease default tolerance in this PR?

**Review by KrisThielemans — COMMENTED** [review]
Ideally would have a test for `pTest` but maybe not to easy (intentionally have a test that fails, and if so, the pTesttest succeeds!)

File: src/common/Utilities.py (hunk 1 above)
Comment (evgueni-ovtchinnikov) [reply src/common/Utilities.py:234]: taken from #844

Comment (KrisThielemans) [reply src/common/Utilities.py:234]: ok. confusing to mix the 2 PRs (might generate a conflict) but fine for me.

**Comment by evgueni-ovtchinnikov** [comment]
@KrisThielemans re testing `pTest`: done manually by introducing errors in some test scripts and making sure they are reported by both `ctest` and `run_all.py`, with all good tests run unaffected

**Comment by KrisThielemans** [comment]
> @KrisThielemans re testing pTest: done manually by introducing errors in some test scripts and making sure they are reported by both ctest and run_all.py, with all good tests run unaffected

Not ideal, but certainly good enough for me, given our priority list. Feel free to merge this whenever Travis is happy (enough).

**Pull request merged**